### PR TITLE
Add workspace selection into Reports view

### LINKED ISSUE #2703: Reports: WS select UI
Requires #2702
iOS version here: #2624

### PR DESCRIPTION
**Pull request by kornelijepetak**
## What's this?

User can now select which workspace to show reports for.

Note: The issue is still waiting for design answers, so the briefcase icon is not white as it is supposed to be.

### Relationships
Closes #2703

## Why do we want this?
Because otherwise user can't show reports for other workspaces other than default.

## How is it done?
The FAB is used here to invoke the selection dialog.

### Why not another way?
Come on! I did it to the best of my abilities.

### Side effects
I don't think so. 

## Review hints
Two things I advice special consideration.

I am afraid of racing condition between:
* current WS index calculation and 
* changes to the collection of workspaces

Not sure how to properly handle this without turning this whole view into RXFactor (which I don't want to do right now).

The other thing is the fact that I'm not sure if current dialog manipulation can cause memory leaks, so please take that into account and raise an alarm if you see something wrong.

## :squid: Permissions
If you are confident enough, feel free.

### CODE REVIEW

**Comment by dshr** [comment]
>I am afraid of racing condition between:
>- current WS index calculation and
>- changes to the collection of workspaces

Tried getting a race condition on this. What I did is throttle my network to Edge, create a new WS, trigger a sync and then try to open the WS selection dialog. Most of the time I got the new WS in there, but even when we're getting a stale collection nothing is broken. I don't think this is something to be bothered about.

> The other thing is the fact that I'm not sure if current dialog manipulation can cause memory leaks, so please take that into account and raise an alarm if you see something wrong.

Code-wise everything seems to work fine. I tried running the Xamarin Profiler on this, but it wouldn't detect my running app. Maybe someone else can try it?

**Comment by kornelijepetak** [comment]
@daividssilverio I've been changing workspace as fast as I could for a full minute. I did not notice anything out of ordinary. With every change, the cards and the piechart changed appropriately. If you can reproduce it consistently, please create another issue.

File: Toggl.Giskard/Resources/layout/ReportsActivity.axml [hunk 1]
```diff
@@ -50,4 +62,14 @@
             android:layout_height="wrap_content"
             android:layout_gravity="start|center_vertical" />
     </android.support.v7.widget.Toolbar>
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/SelectWorkspaceFAB"
+        local:elevation="4dp"
+        android:src="@drawable/workspace_white"
```
Comment (daividssilverio) [inline Toggl.Giskard/Resources/layout/ReportsActivity.axml]: Could you import the svg instead? Let's save some bytes o/

**Comment by daividssilverio** [comment]
@kornelijepetak, I've tried to reproduce the bug I found on my two devices and I can't do it anymore. I guess it might have been the Xamarin version I was using. After you change the resources to SVG, you have my 👍 .

File: Toggl.Giskard/Adapters/ReportsRecyclerAdapter.cs [hunk 2]
```diff
@@ -48,16 +53,27 @@ public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int positi
...
-        public override int ItemCount => 1 + base.ItemCount;
+        public override int ItemCount
+            => workspaceNameCell + summaryCardCell + base.ItemCount;
 
-        public override object GetItem(int viewPosition) 
+        public override object GetItem(int viewPosition)
         {
-            if (viewPosition == 0)
-               return ViewModel;
+            if (viewPosition == ReportsTemplateSelector.WorkspaceName)
```
[the remark this answers is not in the document]
Comment (kornelijepetak) [reply Toggl.Giskard/Adapters/ReportsRecyclerAdapter.cs]: I am aware of what these are. But I agree that it's maybe not a semantically right thing to do. However, I'd rather use variables than magical constants... Will introduce better named variables.

**Comment by daividssilverio** [comment]
I've added a follow up issue for the bug I found but can't easily reproduce #3269

**Pull request merged**